core/src/main/java/org/jdbi/v3/core/statement/Call.java

```diff
@@ -117,31 +129,84 @@ public void invoke(Consumer<OutParameters> resultConsumer) {
      * returning a computed value of type {@code T}.
      */
     public <T> T invoke(Function<OutParameters, T> resultComputer) {
-        try {
-            // it is ok to ignore the PreparedStatement returned here. internalExecute registers it to close with the context and the
-            // nullSafeCleanUp below will take care of it.
-            internalExecute();
-            OutParameters out = new OutParameters(getContext());
-            for (OutParamArgument param : params) {
-                final Object obj = param.map((CallableStatement) stmt);
-
-                // convert from JDBC 1-based position to Jdbi's 0-based
-                final int index = param.position - 1;
-
-                if (param.isNull((CallableStatement) stmt)) {
-                    out.getMap().put(index, null);
-                } else {
-                    out.getMap().put(index, obj);
-                }
+        // it is ok to ignore the PreparedStatement returned here. internalExecute registers it to close with the context and the
+        // nullSafeCleanUp below will take care of it.
+        internalExecute();
 
-                if (param.name != null) {
-                    out.getMap().put(param.name, obj);
-                }
+        final Supplier<ResultSet> resultSetSupplier = () -> {
+            try {
+                return stmt.getResultSet();
+            } catch (SQLException e) {
+                throw Sneaky.throwAnyway(e);
             }
-            return resultComputer.apply(out);
-        } finally {
-            close();
-        }
+        };
+
+        final ResultBearing resultSet = ResultBearing.of(ResultSetSupplier.notClosingContext(resultSetSupplier), getContext());
+
+        OutParameters out = new OutParameters(resultSet, getContext());
+
+        outParamArguments.forEach(outparamArgument -> {
+            Supplier<Object> supplier = outparamArgument.supplier((CallableStatement) stmt);
+            // index is 0 based, position is 1 based.
+            out.putValueSupplier(outparamArgument.position - 1, outparamArgument.name, supplier);
+        });
+
+        return resultComputer.apply(out);
+    }
+
+    /**
+     * Specify the fetch size for the call. This should cause the results to be
```

Comment:
This wording feels a bit stronger than the JDBC spec that backs it, which calls it a "hint" (unlike the other tuneables here)

Reply:
This is cut-and-paste code from Query.